### ISSUE TITLE
removing goid as dependency

### ISSUE THREAD

**Issue by alexmaloteaux** [linkdata/deadlock#1]
Hi , thanks for the revival :)

When compiling with garble im facing an issue with goid

```
# github.com/petermattis/goid
/tmp/garble-shared1926219085/EBih0wwzivaK/NOrHNRTEp.s:29: expected pseudo-register; found R14
asm: assembly of /tmp/garble-shared1926219085/EBih0wwzivaK/NOrHNRTEp.s failed
```

while it looks like a garble issue with build tags, after looking at goid, it looks like since go 1.8 only the slow version is used anyway.

so to fix this i added a goid.go file in deadlock with this code : 

```
package deadlock

import (
        "bytes"
        "runtime"
        "strconv"
)


func ExtractGID(s []byte) int64 {
        s = s[len("goroutine "):]
        s = s[:bytes.IndexByte(s, ' ')]
        gid, _ := strconv.ParseInt(string(s), 10, 64)
        return gid
}

// Parse the goid from runtime.Stack() output. Slow, but it works.
func GetGID() int64 {
        var buf [64]byte
        return ExtractGID(buf[:runtime.Stack(buf[:], false)])
}
```
and replace the goid.xx function with those one. Like this all works for me and the garble compilation succeed.

Dunno if you would be interested to also add it to this fork.

Best Regards

**Comment by linkdata** [comment]
> after looking at goid, it looks like since go 1.8 only the slow version is used anyway.

Did you verify this? If true, then we should remove that package as a dependency.

**Comment by linkdata** [comment]
Fixed in v0.5.1. Use the build tag 'slowgoid' to parse the stack traces instead of using the fast function.

We still want to use github.com/petermattis/goid because it's an order of magnitute faster than getting and parsing the stack trace, but this package will now check that the fast function works correctly at startup and panic if it doesn't.
